lib/api/core/validation/types/date.js

```diff
@@ -221,11 +227,11 @@ function convertRangeValue(name, value) {
   } else if (typeof value === 'number') { // epoch or epoch-millis
     converted = moment.utc(value);
   } else {
-    throw new PreconditionError(`Option "range.${name}": invalid format`);
+    errorsManager.throw('internal', 'validation', 'invalid_range_format', `range.${name}`);
   }
 
   if (!converted.isValid()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "converted" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2259)